src/oas/loader.ts

```diff
@@ -1,19 +1,85 @@
+import OasLoaderInterface from './oas-loader-interface';
 import { API } from './../ms3/ms3-v1-api-interface';
 import LoaderInterface from './../common/loader-interface';
 import * as fs from 'fs';
 import * as util from 'util';
+import * as unzip from 'unzip';
 import { oas30Interface, oas20Interface } from '../oas/index';
 import * as YAML from 'yamljs';
+import { findLastIndex } from 'lodash';
+import Finder from 'node-find-files2';
+import * as rmdir from 'rmdir';
 
+const rmdirPromise = util.promisify(rmdir);
 const readFilePromise = util.promisify(fs.readFile);
 
 export default class OasLoader implements LoaderInterface {
   constructor(private path: string = '') {
     if (!path) throw new Error('Empty path');
   }
 
-  private parseZip() {
+  private parseZip(): Promise<OasLoaderInterface> {
+    return new Promise( (resolve, reject) => {
+      const readStream = fs.createReadStream(this.path);
+      const unzipExtractor = unzip.Extract({ path: './.temp' });
+      readStream.pipe(unzipExtractor);
+      unzipExtractor.on('close', () => {
+        this.findApiJsonFile()
+          .then((api) => {
+            resolve(api);
+          })
+          .catch(function(error) {
+            reject(error);
+          });
+      });
+    });
+  }
+
+  private async findApiJsonFile (): Promise<OasLoaderInterface> {
+    const finder = new Finder({
+      rootFolder : './.temp',
+      filterFunction : (path: string, stat: any) => {
+        const isJsonFile = (path.substring(path.length - 8) == 'api.yaml') ? true : false;
```

Comment:
why only yaml? and fuction is called findApiJsonFile
also if there is no api.yaml or api.json file it should get all files on first level and check which one of them is main file